versions/development/SPEC.md

```diff
@@ -344,20 +345,77 @@ Tasks and workflow inputs may be passed in from an external source, or they may
 
 A string literal may contain any unicode characters between single or double-quotes, with the exception of a few special characters that must be escaped:
 
-| Escape Sequence | Meaning | \x Equivalent | Context |
-|----|---|------|--|
-|`\\`|`\`|`\x5C`||
-|`\n`|newline|`\x0A`||
-|`\t`|tab|`\x09`||
-|`\'`|single quote|`\x27`|within a single-quoted string|
-|`\"`|double quote|`\x22`|within a double-quoted string|
+| Escape Sequence | Meaning      | \x Equivalent | Context                       |
+| --------------- | ------------ | ------------- | ----------------------------- |
+| `\\`            | `\`          | `\x5C`        |                               |
+| `\n`            | newline      | `\x0A`        |                               |
+| `\t`            | tab          | `\x09`        |                               |
+| `\'`            | single quote | `\x27`        | within a single-quoted string |
+| `\"`            | double quote | `\x22`        | within a double-quoted string |
 
 Strings can also contain the following types of escape sequences:
 
 * An octal escape code starts with `\`, followed by 3 digits of value 0 through 7 inclusive.
 * A hexadecimal escape code starts with `\x`, followed by 2 hexadecimal digits `0-9a-fA-F`. 
 * A unicode code point starts with `\u` followed by 4 hexadecimal characters or `\U` followed by 8 hexadecimal characters `0-9a-fA-F`.
 
+##### Multi-line Strings
+
+Strings that begin and end with three consecutive single- or double-quotes may span multiple lines. The opening quotes of a multi-line string may be followed by whitespace (optional) and must have a newline before the first non-whitespace character; these leading whitespace/newline characters are removed. All subsequent non-empty lines are then used to determine the multi-line string's *common leading whitespace* - the minimum number of whitespace characters occuring before the first non-whitespace character in a line or the end of the line, whichever comes first. This common leading whitespace is stripped from the beginning of all the lines in the multi-line string.
+
+```wdl
+# These three strings are equivalent. The middle line of each is empty and so does not count
+# towards the common leading whitespace determination.
+
+String multi_line_A = '''
+        this is a
+
+          multi-line string'''
+
+String multi_line_B = """
+  this is a
+
+    multi-line string"""
+
+String multi_line_C = """
+this is a
+
+  multi-line string"""
+```
+
```

Comment:
I think it would be good add an example where the first line is longer than other lines:

```suggestion
String multi_line_D = """
    this is a
  a multi-line string"""
```